storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -137,14 +138,11 @@
   check highWaterMark and lowWaterMark for backpressure"
   (disruptor/disruptor-backpressure-handler
     (fn []
-      "When worker's queue is above highWaterMark, we set its backpressure flag"
-      (if (not @(:backpressure worker))
-        (do (reset! (:backpressure worker) true)
-            (WorkerBackpressureThread/notifyBackpressureChecker (:backpressure-trigger worker)))))  ;; set backpressure no matter how the executors are
+       (reset! (:transfer-backpressure worker) true)
+       (WorkerBackpressureThread/notifyBackpressureChecker (:backpressure-trigger worker)))
```

Comment:
Never mind for this concern. Because I find that a new flag "_throttleOn" is added to each DisruptorQueue, therefore, we avoid redundant callings of highWaterMark and lowWaterMark functions. So this should work very well!